lib/mri/uri/generic.rb

```diff
@@ -591,7 +591,7 @@ def check_host(v)
       if @opaque
         raise InvalidURIError,
           "can not set host with registry or opaque"
-      elsif parser.regexp[:HOST] !~ v
+      elsif !parser.regexp[:HOST].match v
```

Comment:
> This will also be overwritten on the next MRI stdlib update.

But if you didn't know, the update system does mean it's easy to see this and put the changes back in. That's what I plan to do with `openssl`.